ambry-cloud/src/main/java/com.github.ambry.cloud/azure/AzureBlobDataAccessor.java

```diff
@@ -364,24 +363,56 @@ private BlobContainerClient getContainer(String containerName, boolean autoCreat
   }
 
   /**
-   * @return the name of the Azure storage container where blobs in the specified partition are stored.
-   * @param partitionPath the lexical path of the Ambry partition.
+   * @return the name of the Azure storage container to store the specified blob.
+   * @param blobId the id of the blob to store.
    */
-  private String getAzureContainerName(String partitionPath) {
+  String getAzureContainerName(BlobId blobId) {
```

Comment:
Looks like we are passing blobid to this method now (instead of partitionID as earlier), and then creating CloudBlobMetadata object out of it, but only really using partitionId eventually. Unless I missed something, do we really need these changes?

Reply:
This is to support deriving the Azure container name from either the blob's partition (for backup) or the account/container (for live).